// @fluentui/react-table/lib-commonjs/components/TableBody/useTableBodyStyles.styles.raw.js
'use client';
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    tableBodyClassName: function() {
        return tableBodyClassName;
    },
    tableBodyClassNames: function() {
        return tableBodyClassNames;
    },
    useTableBodyStyles_unstable: function() {
        return useTableBodyStyles_unstable;
    }
});
const _react = require("@griffel/react");
const useTableLayoutStyles = (0, _react.makeStyles)({
    root: {
        display: 'table-row-group'
    }
});
const useFlexLayoutStyles = (0, _react.makeStyles)({
    root: {
        display: 'block'
    }
});
const tableBodyClassName = 'fui-TableBody';
const tableBodyClassNames = {
    root: 'fui-TableBody'
};
const useTableBodyStyles_unstable = (state)=>{
    'use no memo';
    const layoutStyles = {
        table: useTableLayoutStyles(),
        flex: useFlexLayoutStyles()
    };
    state.root.className = (0, _react.mergeClasses)(tableBodyClassName, state.noNativeElements ? layoutStyles.flex.root : layoutStyles.table.root, state.root.className);
    return state;
};
