// @fluentui/react-teaching-popover/lib-commonjs/components/TeachingPopoverSurface/useTeachingPopoverSurfaceStyles.styles.raw.js
'use client';
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    teachingPopoverSurfaceClassNames: function() {
        return teachingPopoverSurfaceClassNames;
    },
    useTeachingPopoverSurfaceStyles_unstable: function() {
        return useTeachingPopoverSurfaceStyles_unstable;
    }
});
const _react = require("@griffel/react");
const _reactpopover = require("@fluentui/react-popover");
const _reacttheme = require("@fluentui/react-theme");
const teachingPopoverSurfaceClassNames = {
    root: 'fui-TeachingPopoverSurface'
};
const useStyles = (0, _react.makeStyles)({
    root: {
        padding: `${_reacttheme.tokens.spacingVerticalL} ${_reacttheme.tokens.spacingVerticalL}`,
        borderRadius: _reacttheme.tokens.borderRadiusXLarge,
        minWidth: '320px',
        boxSizing: 'border-box'
    }
});
const useTeachingPopoverSurfaceStyles_unstable = (state)=>{
    'use no memo';
    const styles = useStyles();
    // Make sure to merge teaching bubble surface prior to popover styles
    state.root.className = (0, _react.mergeClasses)(teachingPopoverSurfaceClassNames.root, styles.root, state.root.className);
    const updatedState = (0, _reactpopover.usePopoverSurfaceStyles_unstable)(state);
    return updatedState;
};
