// @fluentui/react-nav/lib-commonjs/components/NavDrawerHeader/useNavDrawerHeaderStyles.styles.raw.js
'use client';
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    navDrawerHeaderClassNames: function() {
        return navDrawerHeaderClassNames;
    },
    useNavDrawerHeaderStyles_unstable: function() {
        return useNavDrawerHeaderStyles_unstable;
    }
});
const _react = require("@griffel/react");
const _reactdrawer = require("@fluentui/react-drawer");
const navDrawerHeaderClassNames = {
    root: 'fui-NavDrawerHeader'
};
/**
 * Styles for the root slot
 */ const useStyles = (0, _react.makeStyles)({
    root: {
        margin: 'unset',
        paddingInlineStart: '14px',
        paddingBlock: '5px'
    }
});
const useNavDrawerHeaderStyles_unstable = (state)=>{
    'use no memo';
    (0, _reactdrawer.useDrawerHeaderStyles_unstable)(state);
    const styles = useStyles();
    state.root.className = (0, _react.mergeClasses)(navDrawerHeaderClassNames.root, styles.root, state.root.className);
    return state;
};
