// @fluentui/react-menu/lib-commonjs/utils/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    MENU_ENTER_EVENT: function() {
        return _useOnMenuEnter.MENU_ENTER_EVENT;
    },
    MENU_SAFEZONE_TIMEOUT_EVENT: function() {
        return _useOnMenuSafeZoneTimeout.MENU_SAFEZONE_TIMEOUT_EVENT;
    },
    dispatchMenuEnterEvent: function() {
        return _useOnMenuEnter.dispatchMenuEnterEvent;
    },
    useIsSubmenu: function() {
        return _useIsSubmenu.useIsSubmenu;
    },
    useOnMenuMouseEnter: function() {
        return _useOnMenuEnter.useOnMenuMouseEnter;
    },
    useOnMenuSafeZoneTimeout: function() {
        return _useOnMenuSafeZoneTimeout.useOnMenuSafeZoneTimeout;
    },
    useValidateNesting: function() {
        return _useValidateNesting.useValidateNesting;
    }
});
const _useOnMenuEnter = require("./useOnMenuEnter");
const _useIsSubmenu = require("./useIsSubmenu");
const _useValidateNesting = require("./useValidateNesting");
const _useOnMenuSafeZoneTimeout = require("./useOnMenuSafeZoneTimeout");
