// @fluentui/react-nav/lib-commonjs/components/NavDrawerFooter/useNavDrawerFooterStyles.styles.raw.js
'use client';
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    navDrawerFooterClassNames: function() {
        return navDrawerFooterClassNames;
    },
    useNavDrawerFooterStyles_unstable: function() {
        return useNavDrawerFooterStyles_unstable;
    }
});
const _react = require("@griffel/react");
const _reacttheme = require("@fluentui/react-theme");
const _reactdrawer = require("@fluentui/react-drawer");
const navDrawerFooterClassNames = {
    root: 'fui-NavDrawerFooter'
};
/**
 * Styles for the root slot
 */ const useStyles = (0, _react.makeStyles)({
    root: {
        padding: `${_reacttheme.tokens.spacingVerticalXXS} ${_reacttheme.tokens.spacingHorizontalXS} ${_reacttheme.tokens.spacingVerticalXXS} ${_reacttheme.tokens.spacingVerticalXS}  ${_reacttheme.tokens.spacingHorizontalMNudge}`,
        display: 'flex',
        flexDirection: 'column',
        rowGap: _reacttheme.tokens.spacingVerticalXXS
    }
});
const useNavDrawerFooterStyles_unstable = (state)=>{
    'use no memo';
    (0, _reactdrawer.useDrawerFooterStyles_unstable)(state);
    const styles = useStyles();
    state.root.className = (0, _react.mergeClasses)(navDrawerFooterClassNames.root, styles.root, state.root.className);
    return state;
};
