// @fluentui/react-radio/lib-commonjs/components/RadioGroup/useRadioGroupStyles.styles.raw.js
'use client';
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    radioGroupClassNames: function() {
        return radioGroupClassNames;
    },
    useRadioGroupStyles_unstable: function() {
        return useRadioGroupStyles_unstable;
    }
});
const _react = require("@griffel/react");
const radioGroupClassNames = {
    root: 'fui-RadioGroup'
};
const useStyles = (0, _react.makeStyles)({
    root: {
        display: 'flex',
        alignItems: 'flex-start'
    },
    vertical: {
        flexDirection: 'column'
    }
});
const useRadioGroupStyles_unstable = (state)=>{
    'use no memo';
    const styles = useStyles();
    state.root.className = (0, _react.mergeClasses)(radioGroupClassNames.root, styles.root, state.layout === 'vertical' && styles.vertical, state.root.className);
    return state;
};
