// @fluentui/react-badge/lib-commonjs/components/CounterBadge/useCounterBadgeStyles.styles.raw.js
'use client';
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    counterBadgeClassNames: function() {
        return counterBadgeClassNames;
    },
    useCounterBadgeStyles_unstable: function() {
        return useCounterBadgeStyles_unstable;
    }
});
const _react = require("@griffel/react");
const _useBadgeStylesstyles = require("../Badge/useBadgeStyles.styles");
const counterBadgeClassNames = {
    root: 'fui-CounterBadge',
    icon: 'fui-CounterBadge__icon'
};
const useStyles = (0, _react.makeStyles)({
    dot: {
        minWidth: 'auto',
        width: '6px',
        height: '6px',
        padding: '0'
    },
    hide: {
        display: 'none'
    }
});
const useCounterBadgeStyles_unstable = (state)=>{
    'use no memo';
    const styles = useStyles();
    state.root.className = (0, _react.mergeClasses)(counterBadgeClassNames.root, state.dot && styles.dot, !state.root.children && !state.dot && styles.hide, state.root.className);
    if (state.icon) {
        state.icon.className = (0, _react.mergeClasses)(counterBadgeClassNames.icon, state.icon.className);
    }
    return (0, _useBadgeStylesstyles.useBadgeStyles_unstable)(state);
};
