// @fluentui/react-teaching-popover/lib-commonjs/components/TeachingPopoverCarouselPageCount/useTeachingPopoverCarouselPageCountStyles.styles.raw.js
'use client';
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    teachingPopoverCarouselPageCountClassNames: function() {
        return teachingPopoverCarouselPageCountClassNames;
    },
    useTeachingPopoverCarouselPageCountStyles_unstable: function() {
        return useTeachingPopoverCarouselPageCountStyles_unstable;
    }
});
const _react = require("@griffel/react");
const teachingPopoverCarouselPageCountClassNames = {
    root: 'fui-TeachingPopoverCarouselPageCount'
};
/**
 * Styles for the root slot
 */ const useStyles = (0, _react.makeStyles)({
    root: {
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center'
    }
});
const useTeachingPopoverCarouselPageCountStyles_unstable = (state)=>{
    'use no memo';
    const styles = useStyles();
    state.root.className = (0, _react.mergeClasses)(teachingPopoverCarouselPageCountClassNames.root, styles.root, state.root.className);
    return state;
};
