// @fluentui/react-nav/lib-commonjs/components/NavDrawerBody/useNavDrawerBodyStyles.styles.raw.js
'use client';
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    navDrawerBodyClassNames: function() {
        return navDrawerBodyClassNames;
    },
    useNavDrawerBodyStyles_unstable: function() {
        return useNavDrawerBodyStyles_unstable;
    }
});
const _react = require("@griffel/react");
const _reacttheme = require("@fluentui/react-theme");
const _reactdrawer = require("@fluentui/react-drawer");
const navDrawerBodyClassNames = {
    root: 'fui-NavDrawerBody'
};
/**
 * Styles for the root slot
 */ const useStyles = (0, _react.makeStyles)({
    root: {
        padding: `0 ${_reacttheme.tokens.spacingHorizontalXS} 0 ${_reacttheme.tokens.spacingHorizontalMNudge}`,
        alignItems: 'unset',
        display: 'flex',
        flexDirection: 'column',
        rowGap: _reacttheme.tokens.spacingVerticalXXS
    }
});
const useNavDrawerBodyStyles_unstable = (state)=>{
    'use no memo';
    (0, _reactdrawer.useDrawerBodyStyles_unstable)(state);
    const styles = useStyles();
    state.root.className = (0, _react.mergeClasses)(navDrawerBodyClassNames.root, styles.root, state.root.className);
    return state;
};
