// @fluentui/react-table/lib-commonjs/components/TableCellActions/useTableCellActionsStyles.styles.raw.js
'use client';
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    tableCellActionsClassNames: function() {
        return tableCellActionsClassNames;
    },
    useTableCellActionsStyles_unstable: function() {
        return useTableCellActionsStyles_unstable;
    }
});
const _react = require("@griffel/react");
const tableCellActionsClassNames = {
    root: 'fui-TableCellActions'
};
/**
 * Styles for the root slot
 */ const useStyles = (0, _react.makeStyles)({
    root: {
        backgroundColor: 'inherit',
        position: 'absolute',
        right: '0px',
        top: '50%',
        transform: 'translateY(-50%)',
        opacity: 0,
        marginLeft: 'auto'
    },
    visible: {
        opacity: 1
    }
});
const useTableCellActionsStyles_unstable = (state)=>{
    'use no memo';
    const styles = useStyles();
    state.root.className = (0, _react.mergeClasses)(tableCellActionsClassNames.root, styles.root, state.visible && styles.visible, state.root.className);
    return state;
};
