// @fluentui/react-menu/lib-commonjs/components/MenuList/useMenuListStyles.styles.raw.js
'use client';
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    menuListClassNames: function() {
        return menuListClassNames;
    },
    useMenuListStyles_unstable: function() {
        return useMenuListStyles_unstable;
    }
});
const _react = require("@griffel/react");
const menuListClassNames = {
    root: 'fui-MenuList'
};
const useStyles = (0, _react.makeStyles)({
    root: {
        display: 'flex',
        flexDirection: 'column',
        gap: '2px'
    },
    hasMenuContext: {
        height: '100%'
    }
});
const useMenuListStyles_unstable = (state)=>{
    'use no memo';
    const styles = useStyles();
    state.root.className = (0, _react.mergeClasses)(menuListClassNames.root, styles.root, state.hasMenuContext && styles.hasMenuContext, state.root.className);
    return state;
};
