// @fluentui/react-nav/lib-commonjs/components/Nav/useNavStyles.styles.raw.js
'use client';
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    navClassNames: function() {
        return navClassNames;
    },
    useNavStyles_unstable: function() {
        return useNavStyles_unstable;
    }
});
const _react = require("@griffel/react");
const navClassNames = {
    root: 'fui-Nav'
};
/**
 * Styles for the root slot
 */ const useStyles = (0, _react.makeStyles)({
    root: {
        display: 'flex',
        flexDirection: 'column'
    }
});
const useNavStyles_unstable = (state)=>{
    'use no memo';
    const styles = useStyles();
    state.root.className = (0, _react.mergeClasses)(navClassNames.root, styles.root, state.root.className);
    // TODO Add class names to slots, for example:
    // state.mySlot.className = mergeClasses(styles.mySlot, state.mySlot.className);
    return state;
};
