// @fluentui/react-menu/lib-commonjs/components/MenuGroupHeader/useMenuGroupHeaderStyles.styles.raw.js
'use client';
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    menuGroupHeaderClassNames: function() {
        return menuGroupHeaderClassNames;
    },
    useMenuGroupHeaderStyles_unstable: function() {
        return useMenuGroupHeaderStyles_unstable;
    }
});
const _react = require("@griffel/react");
const _reacttheme = require("@fluentui/react-theme");
const menuGroupHeaderClassNames = {
    root: 'fui-MenuGroupHeader'
};
const useStyles = (0, _react.makeStyles)({
    root: {
        fontSize: _reacttheme.tokens.fontSizeBase200,
        color: _reacttheme.tokens.colorNeutralForeground3,
        paddingLeft: '8px',
        paddingRight: '8px',
        fontWeight: _reacttheme.tokens.fontWeightSemibold,
        height: '32px',
        display: 'flex',
        alignItems: 'center'
    }
});
const useMenuGroupHeaderStyles_unstable = (state)=>{
    'use no memo';
    const styles = useStyles();
    state.root.className = (0, _react.mergeClasses)(menuGroupHeaderClassNames.root, styles.root, state.root.className);
    return state;
};
