// @fluentui/react-breadcrumb/lib-commonjs/components/BreadcrumbItem/useBreadcrumbItemStyles.styles.raw.js
'use client';
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    breadcrumbItemClassNames: function() {
        return breadcrumbItemClassNames;
    },
    useBreadcrumbItemStyles_unstable: function() {
        return useBreadcrumbItemStyles_unstable;
    }
});
const _react = require("@griffel/react");
const _reacttheme = require("@fluentui/react-theme");
const breadcrumbItemClassNames = {
    root: 'fui-BreadcrumbItem'
};
const useBreadcrumbItemResetStyles = (0, _react.makeResetStyles)({
    display: 'flex',
    alignItems: 'center',
    color: _reacttheme.tokens.colorNeutralForeground2,
    boxSizing: 'border-box',
    textWrap: 'nowrap'
});
const useBreadcrumbItemStyles_unstable = (state)=>{
    'use no memo';
    const resetStyles = useBreadcrumbItemResetStyles();
    state.root.className = (0, _react.mergeClasses)(breadcrumbItemClassNames.root, resetStyles, state.root.className);
    return state;
};
